gradle-plugin/kubernetes/src/main/java/org/eclipse/jkube/gradle/plugin/task/KubernetesApplyTask.java

```diff
@@ -28,6 +45,60 @@ public KubernetesApplyTask(Class<? extends KubernetesExtension> extensionClass)
 
   @Override
   public void run() {
-    throw new UnsupportedOperationException("To be implemented");
+    try (KubernetesClient kubernetes = jKubeServiceHub.getClient()) {
+      applyService = jKubeServiceHub.getApplyService();
+
+      final File manifest = kubernetesExtension.getManifest(kitLogger, kubernetes, javaProject);
+      if (!manifest.exists() || !manifest.isFile()) {
+        if (kubernetesExtension.getFailOnNoKubernetesJsonOrDefault()) {
+          throw new IllegalStateException("No such generated manifest file: " + manifest);
+        } else {
+          kitLogger.warn("No such generated manifest file %s for this project so ignoring", manifest);
+          return;
+        }
+      }
+
+      URL masterUrl = kubernetes.getMasterUrl();
+      KubernetesResourceUtil.validateKubernetesMasterUrl(masterUrl);
+      List<HasMetadata> entities = KubernetesHelper.loadResources(manifest);
+      kitLogger.info("Using %s at %s in namespace %s with manifest %s ", OpenshiftHelper.isOpenShift(kubernetes) ? "OpenShift" : "Kubernetes", masterUrl, clusterAccess.getNamespace(), manifest);
+
+      configureApplyService();
+
+      // Apply rest of the entities present in manifest
+      applyEntities(manifest.getName(), entities);
+      kitLogger.info("HINT: Use the command `%s get pods -w` to watch your pods start up", clusterAccess.isOpenShift() ? "oc" : "kubectl");
+    } catch (KubernetesClientException e) {
+      KubernetesResourceUtil.handleKubernetesClientException(e, kitLogger);
+    } catch (IOException ioException) {
+      kitLogger.error("Error in loading Kubernetes Manifests ", ioException);
+      throw new IllegalStateException(ioException);
+    } catch (InterruptedException interruptedException) {
+      Thread.currentThread().interrupt();
+      throw new IllegalStateException(interruptedException.getMessage(), interruptedException);
+    }
+  }
+
+  private void applyEntities(String fileName, final Collection<HasMetadata> entities) throws InterruptedException {
+    KitLogger serviceLogger = createLogger(" [SVC] ");
```

Comment:
Why is this different to what we do in ApplyMojo (`"[[G]][SVC][[G]] [[s]]"`)?

Reply:
I trimmed it since we don't support ansi format specifiers #867

Reply:
Please refer to our earlier discussion:
https://github.com/eclipse/jkube/pull/902#pullrequestreview-752505080

We will address that elsewhere. Preserve it for now, else if we fix the ANSI issue we might miss this.

Reply:
okay